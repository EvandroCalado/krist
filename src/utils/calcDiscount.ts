export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
