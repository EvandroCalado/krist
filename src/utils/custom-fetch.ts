import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
