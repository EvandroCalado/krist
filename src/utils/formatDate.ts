export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
