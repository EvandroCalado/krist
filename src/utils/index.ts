export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
