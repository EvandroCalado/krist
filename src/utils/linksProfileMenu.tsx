import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
