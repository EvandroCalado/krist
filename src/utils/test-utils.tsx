import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
