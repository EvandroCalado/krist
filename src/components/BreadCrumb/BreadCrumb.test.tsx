import { screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { BreadCrumb } from '.';

describe('BreadCrumb', () => {
  it('should render default', () => {
    customRender(<BreadCrumb productName="camiseta" />);

    const linkHome = screen.getByRole('link', { name: 'home' });
    const linkShop = screen.getByRole('link', { name: 'shop' });
    const span = screen.getByText('camiseta');

    expect(linkHome).toBeInTheDocument();
    expect(linkHome).toHaveAttribute('href', '/');
    expect(linkShop).toBeInTheDocument();
    expect(linkShop).toHaveAttribute('href', '/shop');
    expect(span).toBeInTheDocument();
    expect(span).toHaveTextContent('camiseta');
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<BreadCrumb productName="camiseta" />);

    expect(container).toMatchSnapshot();
  });
});
