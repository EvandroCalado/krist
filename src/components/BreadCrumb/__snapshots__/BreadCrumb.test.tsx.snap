// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb should render a snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: .8rem;
  padding: 2.4rem 0;
  text-transform: capitalize;
  font-size: 1.2rem;
  font-weight: 600;
}

.c0>a {
  display: flex;
  align-items: center;
}

.c0>a:hover {
  text-decoration: underline;
}

.c0>a>svg {
  width: 1.5rem;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/"
    >
      home 
      <svg
        class="lucide lucide-chevron-right "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </a>
    <a
      href="/shop"
    >
      shop 
      <svg
        class="lucide lucide-chevron-right "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </a>
    <span>
      camiseta
    </span>
  </div>
  ,
</div>
`;
